### PR TITLE
Feature/bid on item

### PR DESCRIPTION
**Pull request by serpient**
# Things Done
- Fix remaining bugs
- Update bid cards to render green when the auction has ended, and the user has won the bid. 

Screen Shot 2019-08-05 at 3 31 24 PM

### CODE REVIEW

File: src/components/bid_card/BidCard.js [hunk 1]
```diff
@@ -3,14 +3,26 @@ import { dateFormatter } from 'utilities/date';
...
-    <div className='bid--container-info'>
-      <h2>{userName} Bid</h2>
-      <h5>{dateFormatter(timestamp)}</h5>
-    </div>
-    <h1 className='bid--container-price'>${convertPenniesToDollars(bid_price)}</h1>
-  </section>
-)
+const BidCard = ({ timestamp, bid_price, userName, white, winner }) => {
+  const bidClass = () => {
+    if (white && !winner) {
```
Comment (aaizuss) [inline src/components/bid_card/BidCard.js:8]: Is white the default color? Why do you need a white prop?

File: src/components/bid_card/BidCard.js [hunk 2]
```diff
@@ -3,14 +3,26 @@ import { dateFormatter } from 'utilities/date';
...
+    }
+  }
+
+  const bidMessage = winner ? "Won" : "Bid";
+
+  return (
+    <section className={`bid--container ${bidClass()}`}>
+      <div className='bid--container-info'>
+        <h2>{userName} {bidMessage}</h2>
+        <h5>{dateFormatter(timestamp)}</h5>
```
Comment (aaizuss) [inline src/components/bid_card/BidCard.js]: nitpicky, but this should probably be called `formatDate`. I like naming functions as verbs - dateFormatter seems more like the name of an object

File: src/components/item_bids/ItemBids.js [hunk 3]
```diff
@@ -14,16 +14,17 @@ const ItemBids = ({ item_uuid, apiHandler, handleError, fetchBids }) => {
...
-    })}
-  </>
+      <h2 className='header--secondary'>Bids</h2>
+      {bids.map(bid => {
+        return <BidCard
+          white={true}
+          key={bid.uuid}
+          timestamp={bid.timestamp}
+          bid_price={bid.bid_price}
+          userName={"Anonymous"}
```
Comment (aaizuss) [inline src/components/item_bids/ItemBids.js:24]: Why is this hardcoded to be `"Anonymous"`? Are you waiting for some backend work to be completed?

File: src/components/item_page/Item.js [hunk 4]
```diff
@@ -1,33 +1,47 @@
...
   const [error, setError] = useState(null);
   const [refetch, setRefetch] = useState(false);
+  const [bidSuccess, setBidSuccess] = useState(false);
   const { uuid, user_uuid } = props.match.params;
 
   useEffect(() => {
     const fetchItems = async () => {
       const { data, errors } = await props.apiHandler.get(`/api/users/${user_uuid}/items/${uuid}`);
-      data ? setItem(data) : setError(errors);
+      if (data) {
```
Comment (aaizuss) [inline src/components/item_page/Item.js]: Why'd you move from a ternary to `if/else`?

File: src/components/item_page/ItemPage.js [hunk 5]
```diff
@@ -6,9 +6,11 @@ import { convertPenniesToDollars } from 'utilities/price';
 import ItemBidForm from 'components/item_bids/ItemBidForm';
 import Error from 'components/UI/Error';
 
-export const ItemPage = ({ item, error, children, handleBidSubmit }) => {
+export const ItemPage = ({ item, error, children, handleBidSubmit, success }) => {
   const { title, description, price, end_date, image, current_highest_bid } = item;
-  const minimumPrice = price === current_highest_bid ? price : current_highest_bid + 100;
+  const currentPrice = current_highest_bid ? current_highest_bid : price;
```
Comment (aaizuss) [inline src/components/item_page/ItemPage.js]: You might be able to replace this with `currentPrice = current_highest_bid || price` unless you want it to be possible for `currentPrice` to be 0 (I would imagine that you'd never show a 0 bid though)

**Review by aaizuss — APPROVED** [review]
LGTM - just had a couple of questions/nitpicky suggestions

File: src/components/bid_card/BidCard.js (hunk 1 above)
Comment (serpient) [reply src/components/bid_card/BidCard.js:8]: theres 3 colors:
- blue (default) (users bid page)
- white (Item bid page)
- green (winning bid)

File: src/components/item_bids/ItemBids.js (hunk 3 above)
Comment (serpient) [reply src/components/item_bids/ItemBids.js:24]: We wanted to anonymize the item bids, so we have it hard coded to "Anonymous" for Item Bids. This Bid Card is used for other cases as well, where we inject different messages

File: src/components/item_page/Item.js (hunk 4 above)
Comment (serpient) [reply src/components/item_page/Item.js]: I thought I would need to add more conditions, but ended up not using it. I'll switch back to ternary

File: src/components/item_page/ItemPage.js (hunk 5 above)
Comment (serpient) [reply src/components/item_page/ItemPage.js]: Updating!

**Pull request merged**